document/core/exec/instructions.rst

```diff
@@ -3077,66 +3077,69 @@ on the top of the stack.
+    a. Pop the top element from the stack.
 
  6. Assert: The stack is now either empty, or there is an exception handler on the top of the stack.
 
  7. If the stack is empty, then:
 
-    a. **TODO** *Return a result value representing the uncaught exception (will probably just be the same as 11.a.i. below).*
+    a. Return the uncaught exception :math:`\val^n~(\THROWadm~a)` as a value.
 
-.. todo::
-   After PR #221 is resolved, this step should be filled in with a PR to specify uncaught exception results.
+8. Else:
 
-8. Else there is an :ref:`exception handler <syntax-handler>` :math:`H` on the top of the stack.
+   a. Assert: there is an :ref:`exception handler <syntax-handler>` :math:`H` on the top of the stack.
 
 9. Pop the exception handler :math:`H` from the stack.
 
-10. Assert: :math:`H` is either of the form :math:`\CATCHadm\{a^?~\instr^\ast\}^k` or :math:`\DELEGATEadm\{l\}.`
+10. Assert: :math:`H` is either of the a catching or a delegating :ref:`exception handler <syntax-handler>`.
 
-11. If :math:`H` is of the form :math:`\CATCHadm\{a^?~\instr^\ast\}^k`, then:
+11. If :math:`H` is a catching exception handler, let :math:`\{a^?~\instr^\ast\}^k` be its clauses. Then:
 
     a. If :math:`k = 0`, then:
 
-       i. Push the throw context that we collected so far :math:`\XT[\val^n~(\THROWadm~a)]` onto the stack.
+       i. Put the values :math:`\val^n` back onto the stack.
 
-    b. Else :math:`H` is of the form :math:`\CATCHadm\{a_1^?~\instr^\ast\}\{a'^?~\instr'^\ast\}^\ast`.
+       ii. :ref:`Throw <exec-throwadm>` an exception with tag address :math:`a`.
 
-    c. If :math:`a_1^? = \epsilon`, then:
+    b. Else :math:`H`'s clauses are of the form :math:`\{a_1^?~\instr^\ast\}\{a'^?~\instr'^\ast\}^\ast`.
 
-       i. Push :math:`\CAUGHTadm\{a~\val^n\}` onto the stack.
+    c. If :math:`a_1^? = \epsilon`, then:
 
-       ii. Jump to the start of the instruction sequence :math:`\instr^\ast`.
+       i. :ref:`Enter <exec-caughtadm-enter>` :math:`\instr^\ast` with catch clause holding the caught exception :math:`\{a~\val^n\}`.
 
     d. Else if :math:`a_1^? = a`, then:
 
-       i. Push :math:`\CAUGHTadm\{a~\val^n\}` onto the stack.
+       i. :ref:`Enter <exec-caughtadm-enter>` :math:`\val^n~\instr^\ast` with catch clause holding the caught exception :math:`\{a~\val^n\}`.
 
-       ii. Push the values :math:`\val^n` back to the stack.
+    e. Else:
 
-       iii. Jump to the start of the instruction sequence :math:`\instr^\ast`.
+       i. Let :math:`H'` be the catching exception handler with clauses :math:`\{a'^?~\instr'^\ast\}^\ast`.
 
-    e. Else, repeat step 11 for :math:`H = \CATCHadm\{a'^?~\instr'^\ast\}^\ast`.
+       ii. Repeat step 11 for :math:`H'`.
 
-12. Else the handler :math:`H` has the form :math:`\DELEGATEadm\{l\}`.
+12. Else:
 
-13. Assert: due to :ref:`validation <valid-delegate-admin>`, the stack contains at least :math:`l+1` labels.
+    a. Assert: :math:`H` is a delegating exception handler.
```

Comment:
Same nesting oddity here.

Reply:
Since the other "if" branch does not end with a clear "return", here I am putting the rest of the steps nested under the "Else".